src/context/cart-context.js

```diff
@@ -0,0 +1,18 @@
+import { createContext, useReducer, useContext } from "react";
+import { cartReducer } from "../hooks/cartReducer";
+
+const CartContext = createContext();
+
+const CartProvider = ({ children }) => {
+  const [cartState, cartDispatch] = useReducer(cartReducer, []);
```

Comment:
- It is a good practice to keep the reducer state value as an ``` object ``` because that way our reducer becomes extendable.
- Using an array as a reducer state is not a good practice even though we have only one state value i.e for ```cart```.

```suggestion
  const [cartState, cartDispatch] = useReducer(cartReducer, {cart: []});
```